### PR TITLE
feat: add negative test cases for Kueue integration

### PR DESCRIPTION
**Pull request by kpunwatk**
Adds comprehensive negative test cases for EvalHub Kueue integration to validate error handling and edge cases in the API.
 Added 5 negative test scenarios:
    - **TC-NEG-001**: Job submission with non-existent queue name
    - **TC-NEG-002**: Job submission without queue spec (backwards compatibility)
    - **TC-NEG-003**: Unauthorized API access returns 401
    - **TC-NEG-004**: Cross-tenant access returns 400/403
    - **TC-NEG-005**: GET request for non-existent job returns 404

# Pull Request

## Related Issues

- Fixes: 
- JIRA: 

## Please review and indicate how it has been tested

- [ ] Locally
- [ ] Jenkins

## Additional Requirements

- [ ] If this PR introduces a new test image, did you create a PR to mirror it in disconnected environment?
- [ ] If this PR introduces new marker(s)/adds a new component, was relevant ticket created to update relevant Jenkins job?

## Summary by CodeRabbit

* **Tests**
  * Added negative integration tests covering: submission to non-existent queues, submissions without a queue specification, unauthorized requests (401), fetching non-existent jobs (404), and forbidden cross-tenant access.
  * Added test setup and teardown that submit jobs for these scenarios and ensure cleanup (hard-delete) after each test to maintain a clean test environment.

### CODE REVIEW

File: tests/ai_safety/evalhub/kueue/test_evalhub_kueue_negative.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+        job_id = data["resource"]["id"]
+
+        # Verify job status reflects the invalid queue
+        status_response = _get_job_status(
+            host=evalhub_kueue_route.host,
+            token=evalhub_kueue_user_token,
+            ca_bundle_file=evalhub_kueue_ca_bundle_file,
+            tenant=evalhub_kueue_namespace.name,
+            job_id=job_id,
+        )
```
Comment (dbasunag) [inline tests/ai_safety/evalhub/kueue/test_evalhub_kueue_negative.py]: This should be a fixture - it should also include the clean up work.

File: tests/ai_safety/evalhub/kueue/test_evalhub_kueue_negative.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+        # Cleanup
+        delete_evalhub_job(
+            host=evalhub_kueue_route.host,
+            token=evalhub_kueue_user_token,
+            ca_bundle_file=evalhub_kueue_ca_bundle_file,
+            tenant=evalhub_kueue_namespace.name,
+            job_id=job_id,
+            hard_delete=True,
+        )
```
Comment (dbasunag) [inline tests/ai_safety/evalhub/kueue/test_evalhub_kueue_negative.py]: if test fails the clean up code would not be reached. It should be in a fixture with use of `yield` so that no matter what the test sees, the setup and clean up would happen.

File: tests/ai_safety/evalhub/kueue/test_evalhub_kueue_negative.py [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        evalhub_kueue_route: Route,
+        evalhub_kueue_ca_bundle_file: str,
+    ) -> None:
+        """TC-NEG-002: Verify job without queue spec runs without Kueue (backwards compatibility).
+
+        Given EvalHub deployed with or without Kueue,
+        When a job is submitted without the queue field,
+        Then the job is accepted (202) and runs without Kueue management.
+        """
+        payload = build_evalhub_job_payload(
```
Comment (dbasunag) [inline tests/ai_safety/evalhub/kueue/test_evalhub_kueue_negative.py]: ditto,

File: tests/ai_safety/evalhub/kueue/test_evalhub_kueue_negative.py [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+        evalhub_kueue_route: Route,
+        evalhub_kueue_ca_bundle_file: str,
+    ) -> None:
+        """TC-NEG-003: Verify unauthorized request returns 401.
+
+        Given EvalHub requires authentication,
+        When a request is made without a valid bearer token,
+        Then the API returns 401 Unauthorized.
+        """
+        payload = build_evalhub_job_payload(
```
Comment (dbasunag) [inline tests/ai_safety/evalhub/kueue/test_evalhub_kueue_negative.py]: Please consider use of parameterization. The test code have similar flow.

File: tests/ai_safety/evalhub/kueue/test_evalhub_kueue_negative.py [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+            token=evalhub_kueue_user_token,
+            ca_bundle_file=evalhub_kueue_ca_bundle_file,
+            tenant=evalhub_kueue_namespace.name,
+            payload=payload,
+        )
+
+        job_id = data["resource"]["id"]
+
+        # Try to access the job from a different (non-existent) tenant
+        import requests
```
Comment (dbasunag) [inline tests/ai_safety/evalhub/kueue/test_evalhub_kueue_negative.py]: Please don't import inside the test code. It should be at the top of the file.

File: tests/ai_safety/evalhub/kueue/test_evalhub_kueue_negative.py [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+            verify=evalhub_kueue_ca_bundle_file,
+            timeout=10,
+        )
+
+        assert response.status_code in (400, 403), (
+            f"Expected 400 or 403 for cross-tenant access, got {response.status_code}: {response.text}"
+        )
+
+        # Cleanup
+        delete_evalhub_job(
```
Comment (dbasunag) [inline tests/ai_safety/evalhub/kueue/test_evalhub_kueue_negative.py]: Same comment as before.

**Review by dbasunag — CHANGES_REQUESTED** [review]

File: tests/ai_safety/evalhub/kueue/test_evalhub_kueue_negative.py (hunk 4 above)
Comment (kpunwatk) [reply tests/ai_safety/evalhub/kueue/test_evalhub_kueue_negative.py]: Done

**Review by dbasunag — APPROVED** [review]

**Review by sheltoncyril — APPROVED** [review]
/lgtm
